### ISSUE TITLE
Package disappeared after update

### ISSUE THREAD

**Issue by ZenFeedbacker** [mechatroner/sublime_rainbow_csv#27]
Hello,

A few days ago rainbow_csv stopped working on Sublime. The package appears in the package list, but it's no longer visible in the View > Syntax list and csv files are not colored. I imagine the package was updated just before it stopped working since the last version came out in the weekend. I tried removing it and reinstalling it but no luck. Any clues on what might have happened?

I have the following versions:
Rainbow CSV v. 2.0.0
Sublime version: 3.2.2, Build 3211
Mac Catalina v.10.15.5

Thanks!

**Comment by mechatroner** [comment]
Hi @ZenFeedbacker ,
Thank you for the bug report!
I believe you are right, the last update broke something for significant fraction of users.
Could you please try to see if the package logs any errors: Click "View" -> "Show Console" and check if the extension logs any errors/exceptions while you do the following: select any character in an open file -> right-click -> Rainbow CSV -> Enable Simple. 

Another unrelated question, do you have "User" package in the sublime Packages folder? Click "Preferences" -> "Browse Packages" to find out if you have it.

**Comment by ZenFeedbacker** [comment]
Hi @mechatroner , thanks for the quick answer!

Unfortunately when I try to do "Enable Simple' in the right click menu, Sublime freezes and I have to force quit it. Nothing appears in the Console before the freeze. If I try to reopen this file, Sublime freezes again, most probably because Simple parsing is enabled for that file. Other files work normally until I try to enable Simple for them too.

As for the second question, yes, there is a folder name User under ~/Library/Application Support/Sublime Text 3

**Comment by mechatroner** [comment]
Thank you, @ZenFeedbacker 
I still can't reproduce the problem, even tried it on an old Mac laptop. Could you please tell me the list of your installed packages (Preferences -> Package Control -> List Packages) while I am trying to debug this. If I don't find anything I will just try to adjust the most suspicious fragments of the source code later today.

**Comment by mechatroner** [comment]
@ZenFeedbacker, Also could you please try to do the same with versions 2.0.1 or 2.0.2 after setting "enable_debug_logging" option to true. Maybe it would manage to log something before freezing up.  I also fixed some minor bugs in these new versions, maybe this could help too.  Also, I am wondering if you could reproduce this problem with a different file and a different separator.

**Comment by mechatroner** [comment]
I just published version 2.0.3, it fixes some more bugs. I must admit I was very sloppy with release 2.0.0. Please let me know if the problem still persists.

**Comment by JnCoe** [comment]
Hi there! I'm having the same issue as well, I believe it started around a week ago. I opened the console and noticed that there was already an error:
`"Unable to open /C/Users/jonas/AppData/Roaming/Sublime Text 3/Packages/User/RainbowCSV.sublime-settings"`

It also freezes for me when doing the "enable simple"

Currently I have the following packages:
Advanced CSV
AutoHotkey
Hexviewer
Makdown extended
Rainbow_csv
RawLineEdit

But I don't think it influenced because I tried a clean install of sublime text with just Rainbow CSV and the problem persisted.
I'm not sure I know how to downgrade, I changed the log setting to true but apparently no file was created after forcing ST to close. Also tried with a different file using different separators, encapsulated or not, but the same happened.

Currently, I'm using the following:
Rainbow CSV v. 2.0.3
Sublime version: 3.2.2, Build 3211
Windows 10 home 1909

But I think this began with  version 2.0.0 as well.

Also, I'm not sure it's relevant but I'm using python 3.7.7 on Anaconda.

**Comment by ZenFeedbacker** [comment]
@mechatroner I've installed 2.0.3 and I'm getting the same behaviour. I enabled enable_debug_logging and tried with another file and with a comma separator, instead of a semicolon but no luck, the program freezes before anything is logged unfortunately .

The packages I've got are the following

- advanced CSV
- Boxy Theme
- BrackerHighlighter
- Compare Side-By-Side
- Exalt
- Expand Selection to Quotes
- Expand Selection to Whitespace
- FileDiffs
- Filter Lines
- Highlight Duplicates
- Indent XML
- Origami
- Package Control
- Pretty JSON
- rainbow_csv
- RawLineEdit
- RegReplace
- Smart Title Case
- Sort Lines(Numerically)
- SublimeLinter-xmllint
- TabsExtra
- Text Pastry

**Comment by ZenFeedbacker** [comment]
TSV files seem to work correctly though. The annotation even appears in the lower right corner, even if I can find it in the syntax list.
<img width="349" alt="Screenshot 2020-07-13 at 2 04 25 PM" src="https://user-images.githubusercontent.com/35770410/87297949-f4c6b300-c511-11ea-85fe-45e470c39886.png">

**Comment by JnCoe** [comment]
> TSV files seem to work correctly though. The annotation even appears in the lower right corner, even if I can find it in the syntax list.
> <img alt="Screenshot 2020-07-13 at 2 04 25 PM" width="349" src="https://user-images.githubusercontent.com/35770410/87297949-f4c6b300-c511-11ea-85fe-45e470c39886.png">

That's funny, I wasn't able to make it work it TSV here. It doesn't even show on the corner

**Comment by mechatroner** [comment]
Guys, I just published a new version 2.1.0 but it won't be available through package control for 3-4 hours. Essentially I reverted part of the logic to pre 2.0.0 state. Although I am still not sure if this would help or not. I also improved the debug logging, now if you enable `"enable_debug_logging": true` in Rainbow CSV package settings you would be able to see the logging output not only in the sublime console but also in `Packages/User/rainbow_csv_debug.log` file. Please let me know what you see there before the editor freezes because I can't reproduce it myself.

**Comment by khaldi-yass** [comment]
Hello Here is the error that I get when trying to open the csv file,
```terminal
Syntax file basename: "Rainbow_CSV_hex_3b_Standard.sublime-syntax", Created: True
Current syntax file: "Packages/Text/Plain text.tmLanguage", New syntax file: "Packages/User/Rainbow_CSV_hex_3b_Standard.sublime-syntax"
New syntax file created: "Packages/User/Rainbow_CSV_hex_3b_Standard.sublime-syntax". Preparing to enable
generating syntax summary
no such context scope:source.scss#.string-double in Packages/MarkdownEditing/Markdown.sublime-syntax
no such context scope:source.scss#.string-single in Packages/MarkdownEditing/Markdown.sublime-syntax
no such context scope:source.scss#.media-query-list in Packages/MarkdownEditing/Markdown.sublime-syntax
no such context scope:source.scss#.media-query-list in Packages/MarkdownEditing/Markdown.sublime-syntax
no such context scope:source.scss#.string-single in Packages/MarkdownEditing/Markdown.sublime-syntax
no such context scope:source.scss#.string-double in Packages/MarkdownEditing/Markdown.sublime-syntax
In callback. Setting rainbow syntax file to: "Packages/User/Rainbow_CSV_hex_3b_Standard.sublime-syntax"
```

Maybe there is a conflict with the MarkdownEditing package?

**Comment by mechatroner** [comment]
@khaldi-yass, Thanks! actually the log file looks OK. So what happens after you open the csv file, does the editor freeze or the file just not getting highlighted?

**Comment by mechatroner** [comment]
The new version 2.1.0 is now available through Package Control. 
Sublime restart or file reopening can be needed to make sure if it works or still fails.

**Comment by JnCoe** [comment]
I was still with the same issue, however i decided to go nuke and uninstalled ST3 using Revo Unistaller, removing all files that remained in the system. Then I reinstalled it and added rainbow csv and it worked. The previous time a did a clean install I just deleted the AppData folders. Don't know why didn't work before.
I decided to check if it was the package Advanced Csv causing some incompatibility, so I installed only it.
Voilá, the problem was back.
Removed Advanced Csv and rainbow was working fine again.

There seems to be some incompatibility between the two with recent versions.
In any case, I use Rainbow much more than Advanced CSV so it's solved for me.

Thank you very much for you time and dedication! =)

**Comment by mechatroner** [comment]
Thank you @JnCoe !
Please let me know if you have any issues with it again. 
I will also open an issue with Sublime Text itself because there is something wrong with the way it handles autogenerated syntax files. In my understanding, they should be no different than the syntax files provided with the package, but apparently, there is a difference and it varies from system to system.

**Comment by ZenFeedbacker** [comment]
@mechatroner I've updated the plugin to 2.1.0 and here's what happens now:

rainbow_csv has reappeared in the syntax menu. When I open a .csv file the syntax selected is '_Plain Text (csv)_'. If I pick one of the options there they work all right; I've tried 3 or 4 of them with different files and there is no problem. If I choose the option Right click > Enable Standard  I get this:
<img width="422" alt="Screenshot 2020-07-14 at 10 03 47 AM" src="https://user-images.githubusercontent.com/35770410/87395043-53933780-c5b9-11ea-917b-f7620a96c4b9.png">
(even though this is a csv file with semicolon operators)

If i pick the Enable Simple from the same menu, Sublime freezes  (with no output on the console)

**Comment by khaldi-yass** [comment]
> @khaldi-yass, Thanks! actually the log file looks OK. So what happens after you open the csv file, does the editor freeze or the file just not getting highlighted?

Yes yesterday the behaviour was: ST3 opened the file but no highlighting occurs, today after the update it works as before. Thank you.

**Comment by mechatroner** [comment]
@ZenFeedbacker, When you use Right click > Enable Standard you need to visually select the separator text with the mouse cursor i.e. any single ',' or ';' character and then do "Right click" -> "Enable Standard". You are probably choosing a different separator e.g. some other character or sequence of character. In this case, Rainbow CSV will try to autogenerate the appropriate syntax file with the separator you selected. Actually, while I was writing this reply I realized that I made a UI mistake and was finally able to reproduce the freezing at least in one case. Turns out if you don't select any separator at all and do "Right click" -> "Enable  Simple" - it will freeze. I will add error handling for this case and publish a patch.

**Comment by mechatroner** [comment]
@ZenFeedbacker I just published version 2.1.1 - it should fix freezing with empty separators. Please let me know if Sublime still freezes in other cases.

**Comment by ZenFeedbacker** [comment]
@mechatroner Yes that was it, I hadn't realised that you had to choose a separator in the right click context menu. This case works too now with v.2.1.1., I get a pop-up that I can't use an empty string as a separator. Everything is alright then, as far as I'm concerned the issue can be closed. 

And thank you for the quick replies and all the work you've been doing in general, this plugin is a life saver when you work all day with csv files. cheers!
